README.md
RealURL extension for TYPO3 CMS
=============

This is version 2.x of the famous RealURL extension for TYPO3 CMS. This version is rewritten from scratch and will work only with TYPO3 6.2 or newer.

Repository, bug reporting, pull requests, etc is handled via https://github.com/dmitryd/typo3-realurl/


ChangeLog
2014-11-25  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #62145: RealURL Backend module does not work well with SSL
	* Fixed bug #61493: Old class is called in the Info module

2014-11-04  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #62442: Wrong cropping of multibyte URLs
	* Fixed bug #58027: Deprecation warning about GeneralUtility::array_merge_recursive_overrule()

2014-10-07  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #60585: Deleting of single entries of decode/encode cache fails sometimes
	* Fixed bug #60952: strip_tags for lookUp_cleanAlias

2014-03-12  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #61881: Pathcache is not cleared when creating a new translation
	* Fixed bug #56778: languageExceptionUids not working with RealURL 1.12.8

2014-02-17  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.12.8

2014-02-11  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #55436: findRootPageId: Infinite Loop in case of circular redirect
	* Fixed bug #52651: Autoconf won't be created when hook is used
	* Increase TYPO3 compatibility version to 6.2

2013-10-09  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #45879: Unset menu items for ReaulURL management (thanks to  Leon Dietsch)
	* Fixed bug #52515: Add hook to post process auto-generated configuration

2013-10-08  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #51801: typolink.linkAccessRestrictedPages has no effect if RealURL is enabled
	* Fixed bug #51287: add cHash on decoding only if there are relevant parameters
	* Fixed bug #49019: Page caching does not work if cHash cache is disabled
	* Fixed bug #48944: Check if the duplicate unique alias is a translation

2013-10-07  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #51471: fatal error in info module in TYPO3 6.2 (rev a52257a56b0cdb17fd3fbda91ab4c75bd3cad33c)

2013-09-25  TYPO3 Security team <security@typo3.org>

	* Security fix for TYPO3-EXT-SA-2013-015
	* Released version 1.12.7

	WARNING! Version 1.12.7 has only changes made prior/on 2013-05-07 plus security fixes. It does not include fixes from later days. Those fixes appear in 1.12.8 release.

2013-07-24  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Use a distinct status string when redirecting from the RealURL
	* Fixed bug #45398: getLanguageVar() returns 0 instead of sys_language_uid (thanks to Andreas Beutel)
	* Fixed bug #41515: Wrong value applied for GET var when using value default (thanks to Helmut Hummel)
	* Fixed bug #48514: 404 if path segment is empty

2013-05-20  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Remove inclusion of old TYPO3 files. Rely on autoloader and compatibility layer now.
	* Fix 'not deleted' in the manual (wrong SQL syntax in examples)

2013-05-07  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fix mixed case extension key
	* Released version 1.12.6

2013-05-06  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.12.5

2013-04-11  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* [BUGFIX] mysqli compatibility issue with TYPO3 CMS 6.1 (thanks to Benni Mack and Oliver Hader)
	* Fixed bug #46782: avoid using loadTCA with TYPO3 >= 6.1
	* Change url decoding: replace rawurldecode() with urldecode()
	* [BUGFIX] Decodefilename adds rawurlencode to only one segment (thanks to Stephan Schuler)

2013-03-13  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.12.4

2013-02-26  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Added information to the manual about calling userFunc for alias processing (thanks to Christian Ludwig)

2013-02-08  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #43816: flock() calls in autoconf file generation prevent NFS hosting
	* Fixed bug #42295: Revitalization of pathcache confused if nothing changed
	* Fixed bug #41477: Pathcache entries are not expired for subpages (thanks to Steffen Gebert)

2013-01-29  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #43599: 404 error if there is a separator in the subpage

2012-08-23  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Do not set cHash if there are no GET variables (sfp 926.1041)

2012-08-21  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18548: Deprecation Warnings with TYPO3 4.7
	* Fixed bug #18575: wrong code for localized title detection
	* Released version 1.12.3

2012-06-29  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18578: 404 error is not displayed if appendMissingSlash is not set

2012-06-25  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18568: insufficient URL conversion of Redirects with umlauts
	* Fixed bug #18527: URLs with a . at the end do not produce 404 error
	* Fixed bug #18437: Wrong ID in shortcut mode "Parent page"
	* Fixed bug #18443: produce fewer results on title lookup
	* Fixed bug #18190: apendMissingSlash handling different if a point is in query parameter
	* Fixed bug #18133: Double quotation if magic_quotes_gpc is on
	* Released version 1.12.2

2012-06-19  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Snowflake task 1570.179: cHash issues with multilanguage domains

2012-05-29  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18551: target redirect path is saved in lower case
	* Fixed bug #18532: cHash is incorrectly recalculated if the postVar is discarded due to the empty segment

2012-05-22  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18477: cHash is wrong when using noMatch=bypass with the last parameter in the postVarSets

2012-04-23  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18462: wrong definition for uid column in tx_realurl_redirects
	* Fixed bug #18479: redirect statistics sometimes are not updated

2012-02-17  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Released version 1.12.1

2012-02-02  Dmitry Dulepov  <dmitry.dulepov@gmail.com>

	* Fixed bug #18447: Many domains in the same installation is not supported by the _DOMAINS

ext_emconf.php
<?php

/*********************************************************************
* Extension configuration file for ext "realurl".
*
* Generated by ext 17-02-2014 12:13 UTC
*
* https://github.com/t3elmar/Ext
*********************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'RealURL: speaking paths for TYPO3',
  'description' => 'Creates nice looking URLs for TYPO3 pages: converts http://example.com/index.phpid=12345&L=2 to http://example.com/path/to/your/page/. Please, ask for free support in TYPO3 mailing lists or contact the maintainer for paid support.',
  'category' => 'fe',
  'shy' => 0,
  'dependencies' => '',
  'conflicts' => 'cooluri,simulatestatic',
  'priority' => '',
  'loadOrder' => '',
  'module' => '',
  'state' => 'stable',
  'internal' => 0,
  'uploadfolder' => 0,
  'createDirs' => '',
  'modify_tables' => 'pages,sys_domain,pages_language_overlay,sys_template',
  'clearCacheOnLoad' => 1,
  'lockType' => '',
  'author' => 'Dmitry Dulepov',
  'author_email' => 'dmitry.dulepov@gmail.com',
  'author_company' => '',
  'CGLcompliance' => '',
  'CGLcompliance_note' => '',
  'version' => '1.12.8',
  '_md5_values_when_last_written' => 'a:19:{s:9:"ChangeLog";s:4:"3408";s:10:"_.htaccess";s:4:"9ab3";s:20:"class.ext_update.php";s:4:"9794";s:20:"class.tx_realurl.php";s:4:"0cea";s:29:"class.tx_realurl_advanced.php";s:4:"f387";s:32:"class.tx_realurl_autoconfgen.php";s:4:"f227";s:28:"class.tx_realurl_tcemain.php";s:4:"2983";s:16:"ext_autoload.php";s:4:"a6ff";s:21:"ext_conf_template.txt";s:4:"c890";s:12:"ext_icon.gif";s:4:"ea80";s:17:"ext_localconf.php";s:4:"e297";s:14:"ext_tables.php";s:4:"7182";s:14:"ext_tables.sql";s:4:"150c";s:17:"locallang_csh.xml";s:4:"369d";s:16:"locallang_db.xml";s:4:"584a";s:14:"doc/manual.sxw";s:4:"3b3d";s:38:"modfunc1/class.tx_realurl_modfunc1.php";s:4:"d3a6";s:41:"modfunc1/class.tx_realurl_pagebrowser.php";s:4:"60b1";s:22:"modfunc1/locallang.xml";s:4:"7e4f";}',
  'constraints' =>
  array (
    'depends' =>
    array (
      'php' => '5.3.2-5.999.999',
      'typo3' => '4.5.0-6.2.999',
      'lowlevel' => '1.0.0-100.0.0'
    ),
    'conflicts' =>
    array (
      'cooluri' => '',
      'simulatestatic' => '',
    ),
    'suggests' =>
    array (
      'static_info_tables' => '2.0.2-',
    ),
  ),
  'suggests' =>
  array (
  ),
  'user' => 'dmitry',
  'comment' => 'TYPO3 6.2. compatibility update',
);

?>